scala-package/core/src/main/scala/org/apache/mxnet/api/java/JavaNDArray.scala

```diff
@@ -1,95 +1,185 @@
-  def ones(shape: Int *): JavaNDArray = NDArray.ones(shape: _*)
-  def ones(ctx: JavaContext, shape: Int *): JavaNDArray = NDArray.ones(ctx, shape: _*)
+  def zeros(shape: JavaShape, ctx: JavaContext, dtype: JavaDType.DType): JavaNDArray
+  = NDArray.zeros(shape, ctx, dtype)
 
-  def full(shape: JavaShape, value: Float, ctx: JavaContext): JavaNDArray = NDArray.full(shape, value, ctx)
+  def zeros(ctx: JavaContext, shape: Array[Int]): JavaNDArray = NDArray.zeros(ctx, shape: _*)
+
+  def ones(shape: JavaShape, ctx: JavaContext, dtype: JavaDType.DType): JavaNDArray
+  = NDArray.ones(shape, ctx, dtype)
+
+  def full(shape: JavaShape, value: Float, ctx: JavaContext): JavaNDArray
+  = NDArray.full(shape, value, ctx)
 
   def power(lhs: JavaNDArray, rhs: JavaNDArray): JavaNDArray = NDArray.power(lhs, rhs)
+
   def power(lhs: JavaNDArray, rhs: Float): JavaNDArray = NDArray.power(lhs, rhs)
+
   def power(lhs: Float, rhs: JavaNDArray): JavaNDArray = NDArray.power(lhs, rhs)
 
   def maximum(lhs: JavaNDArray, rhs: JavaNDArray): JavaNDArray = NDArray.maximum(lhs, rhs)
+
   def maximum(lhs: JavaNDArray, rhs: Float): JavaNDArray = NDArray.maximum(lhs, rhs)
+
   def maximum(lhs: Float, rhs: JavaNDArray): JavaNDArray = NDArray.maximum(lhs, rhs)
 
   def minimum(lhs: JavaNDArray, rhs: JavaNDArray): JavaNDArray = NDArray.minimum(lhs, rhs)
+
   def minimum(lhs: JavaNDArray, rhs: Float): JavaNDArray = NDArray.minimum(lhs, rhs)
+
   def minimum(lhs: Float, rhs: JavaNDArray): JavaNDArray = NDArray.minimum(lhs, rhs)
 
   def equal(lhs: JavaNDArray, rhs: JavaNDArray): JavaNDArray = NDArray.equal(lhs, rhs)
+
   def equal(lhs: JavaNDArray, rhs: Float): JavaNDArray = NDArray.equal(lhs, rhs)
 
   def notEqual(lhs: JavaNDArray, rhs: JavaNDArray): JavaNDArray = NDArray.notEqual(lhs, rhs)
+
   def notEqual(lhs: JavaNDArray, rhs: Float): JavaNDArray = NDArray.notEqual(lhs, rhs)
 
   def greater(lhs: JavaNDArray, rhs: JavaNDArray): JavaNDArray = NDArray.greater(lhs, rhs)
+
   def greater(lhs: JavaNDArray, rhs: Float): JavaNDArray = NDArray.greater(lhs, rhs)
 
   def greaterEqual(lhs: JavaNDArray, rhs: JavaNDArray): JavaNDArray = NDArray.greaterEqual(lhs, rhs)
+
   def greaterEqual(lhs: JavaNDArray, rhs: Float): JavaNDArray = NDArray.greaterEqual(lhs, rhs)
 
   def lesser(lhs: JavaNDArray, rhs: JavaNDArray): JavaNDArray = NDArray.lesser(lhs, rhs)
+
   def lesser(lhs: JavaNDArray, rhs: Float): JavaNDArray = NDArray.lesser(lhs, rhs)
 
   def lesserEqual(lhs: JavaNDArray, rhs: JavaNDArray): JavaNDArray = NDArray.lesserEqual(lhs, rhs)
+
   def lesserEqual(lhs: JavaNDArray, rhs: Float): JavaNDArray = NDArray.lesserEqual(lhs, rhs)
 
-  def array(sourceArr: java.util.List[Float], shape: JavaShape, ctx: JavaContext = null): JavaNDArray
-    = NDArray.array(sourceArr.asScala.toArray, shape, ctx)
+  def array(sourceArr: Array[Float], shape: JavaShape, ctx: JavaContext = null): JavaNDArray
```

Comment:
I don't think that this is right. A java user will not be making a call to this using Array.